### ISSUE TITLE
Support emitting setters for WebIDL attributes that are not `readonly`

### ISSUE THREAD

**Issue by fitzgen** [rustwasm/wasm-bindgen#252]
https://heycam.github.io/webidl/#idl-attributes
https://heycam.github.io/webidl/#dfn-read-only

We should not emit setters for this:

```webidl
interface Foo {
  readonly attribute long bar;
};
```

But we should emit a `fn bar(&self, val: i32)` setter for this:

```webidl
interface Foo {
  attribute long bar;
};
```

**Comment by ohanar** [comment]
This is implemented here: ohanar@2fbd4d9e3489e2b25b983ebf95c5f167737e98ec (based off of my pull request for snake case methods). The signature is `fn set_bar(&self, bar: i32)`, which follows the conventions in the examples more closely (this can be easily changed).

It currently fails the test suite since it now depends upon the `DOMHighResTimeStamp` type (it would also depend on `EventTarget` if nullable types were supported).

**Comment by fitzgen** [comment]
Thanks for the heads up!
